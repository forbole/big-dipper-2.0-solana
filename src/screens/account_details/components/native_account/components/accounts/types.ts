export type AddressesType = string[];

export type AccountsType = {
  stake: AddressesType;
  nonce: AddressesType;
  vote: AddressesType;
  token: AddressesType;
}
