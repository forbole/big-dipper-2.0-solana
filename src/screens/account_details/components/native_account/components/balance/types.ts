export type BalanceType = {
  native: TokenUnit;
  stake: TokenUnit;
  nonce: TokenUnit;
  vote: TokenUnit;
  total: TokenUnit;
}
