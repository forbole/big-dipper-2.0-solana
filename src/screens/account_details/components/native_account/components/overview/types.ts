export type OverviewType = {
  address: string;
}
