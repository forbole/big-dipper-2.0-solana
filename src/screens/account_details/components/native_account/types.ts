export type OverviewType = {
  address: string;
}

export type AccountType = {
  address: string;
}

export type AccountsType = {
  data: {
    [value:number]: AccountType[];
  };
  count: number;
  loading: boolean;
}

export type NativeAccountState = {
  loading: boolean;
  exists: boolean;
  overview: OverviewType;
}
