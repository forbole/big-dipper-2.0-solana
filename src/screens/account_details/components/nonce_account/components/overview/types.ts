export type OverviewType = {
  address: string;
  balance: number;
  authority: string;
  blockhash: string;
  fee: number;
}
