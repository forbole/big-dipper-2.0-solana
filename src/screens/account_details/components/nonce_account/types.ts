export type OverviewType = {
  address: string;
  authority: string;
  balance: TokenUnit;
  blockhash: string;
  fee: number;
}

export type NonceAccountState = {
  loading: boolean;
  overview: OverviewType;
}
