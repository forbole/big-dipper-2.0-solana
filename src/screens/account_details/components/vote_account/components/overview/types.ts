export type OverviewType = {
  address: string;
  balance: string;
  authorizedVoter: string;
  authorizedWithdrawer: string;
  lastTimestamp: string;
  commission: number;
  rootSlot: number;
}
