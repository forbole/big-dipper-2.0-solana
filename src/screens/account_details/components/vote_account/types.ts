export type OverviewType = {
  address: string;
  balance: TokenUnit;
  voter: string;
  withdrawer: string;
}

export type ValidatorProfile = {
  description: string;
  website: string;
  imageUrl: string;
  active: boolean;
  commission: number;
  lastVote: number;
  rootSlot: number;
}

export type VoteAccountState = {
  loading: boolean;
  overview: OverviewType;
  validatorProfile: ValidatorProfile;
}
