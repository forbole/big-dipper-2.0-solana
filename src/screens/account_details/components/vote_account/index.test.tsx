/* eslint-disable */
import React from 'react';
import { createMockClient } from 'mock-apollo-client';
import { ApolloProvider } from '@apollo/client';
import renderer from 'react-test-renderer';
import {
  MockTheme, wait,
} from '@tests/utils';
import { VoteAccountDetailsDocument } from '@graphql/types';
import VoteAccount from '.';

// ==================================
// mocks
// ==================================
jest.mock('next/router', () => ({
  useRouter: () => ({
    query: {
      address: "9XfsbcUJ8qM4esLtMocrMGLqRPkgYNbcmKLebtBzSjjZ",
    },
  }),
}));

jest.mock('./components', () => ({
  Overview: (props) => <div id="Overview" {...props} />,
  Validator: (props) => <div id="Validator" {...props} />,
  SkipRate: (props) => <div id="SkipRate" {...props} />,
}));

const mockData = jest.fn().mockResolvedValue({
  "data": {
    "validator": [
      {
        "address": "44MUR8uuQZZvNBe9qGBud1EtgYBsUndpiYLAae6HYUc1",
        "nativeBalance": {
          "balance": 26858640
        },
        "commission": 100,
        "voter": "F5NgZ5RtJW6fcDTcsYjSo8DgksbWx3mh4Ms2igbhHQTC",
        "withdrawer": "8RLmw25NXPZ2Ry1iRJFtn5fFkWwpVETEXwu5bnHbWcTJ",
        "validatorStatus": null,
        "validatorConfig": null,
        "validatorSkipRate": {
          "epoch": 303,
          "skip": 5,
          "skipRate": 0.046296296296296294,
          "total": 108
        }
      }
    ]
  }
});

// ==================================
// unit tests
// ==================================
describe('screen: VoteAccount', () => {
  it('matches snapshot', async () => {
    const mockClient = createMockClient();
    mockClient.setRequestHandler(
      VoteAccountDetailsDocument,
      mockData,
    );

    let component;
    renderer.act(() => {
      component = renderer.create(
        <ApolloProvider client={mockClient}>
          <MockTheme>
            <VoteAccount />
          </MockTheme>
        </ApolloProvider>,
      );
    });
    await wait();

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
