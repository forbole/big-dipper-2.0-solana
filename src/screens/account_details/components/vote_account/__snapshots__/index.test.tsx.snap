// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen: VoteAccount matches snapshot 1`] = `
Array [
  <div
    className="makeStyles-validator"
    id="Validator"
    validator={
      Object {
        "active": false,
        "address": "9XfsbcUJ8qM4esLtMocrMGLqRPkgYNbcmKLebtBzSjjZ",
        "commission": 100,
        "description": "",
        "imageUrl": "",
        "name": "9XfsbcUJ8qM4esLtMocrMGLqRPkgYNbcmKLebtBzSjjZ",
        "rootDistance": 0,
        "voteDistance": 0,
        "website": "",
      }
    }
  />,
  <div
    className="makeStyles-overview"
    id="Overview"
    overview={
      Object {
        "address": "9XfsbcUJ8qM4esLtMocrMGLqRPkgYNbcmKLebtBzSjjZ",
        "balance": Object {
          "baseDenom": "udaric",
          "displayDenom": "daric",
          "exponent": 6,
          "value": "26.85864",
        },
        "voter": "F5NgZ5RtJW6fcDTcsYjSo8DgksbWx3mh4Ms2igbhHQTC",
        "withdrawer": "8RLmw25NXPZ2Ry1iRJFtn5fFkWwpVETEXwu5bnHbWcTJ",
      }
    }
  />,
  <div
    className="makeStyles-skipRate"
    id="SkipRate"
    skipRate={
      Object {
        "epoch": 303,
        "skip": 5,
        "skipRate": 0.046296296296296294,
        "total": 108,
      }
    }
  />,
]
`;
